docs/proposal/kmesh_support_workload.md

```diff
@@ -0,0 +1,141 @@
+  // For Kubernetes，this would be <name>.<namespace>.svc.<cluster domain>, such as: "fortio-server.default.svc.cluster.local", "istiod.istio-system.svc.cluster.local" etc.
+  Hostname string `protobuf:"bytes,3,opt,name=hostname,proto3" json:"hostname,omitempty"`
+  // Address represents the addresses the service can be reached at.
+  // There may be multiple addresses for a single service if it resides in multiple networks,
+  // multiple clusters, and/or if it's dual stack (TODO: support dual stack).
+  // For a headless kubernetes service, this list will be empty.
+  Addresses []*NetworkAddress `protobuf:"bytes,4,rep,name=addresses,proto3" json:"addresses,omitempty"`
+  // Ports for the service.
+  // The target_port may be overridden on a per-workload basis.
+  Ports []*Port `protobuf:"bytes,5,rep,name=ports,proto3" json:"ports,omitempty"`
+}
+```
+
+### Workload
+
+```go
+type Workload struct {
+  // UID represents a globally unique opaque identifier for this workload, such as: "Kubernetes//Pod/default/fortio-server-deployment-59f95d774d-85nr4"
+  Uid string `protobuf:"bytes,20,opt,name=uid,proto3" json:"uid,omitempty"`
+  // Name represents the name for the workload, For Kubernetes, this is the pod name, such as: "fortio-server-deployment-59f95d774d-ljmd5"
+  Name string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
+  // Namespace represents the namespace for the workload. 
+  Namespace string `protobuf:"bytes,2,opt,name=namespace,proto3" json:"namespace,omitempty"`
+  // Address represents the IPv4/IPv6 address for the workload, this should be globally unique.
+  Addresses [][]byte `protobuf:"bytes,3,rep,name=addresses,proto3" json:"addresses,omitempty"`
+  // The services for which this workload is an endpoint. the key is the NamespacedHostname string of the format namespace/hostname.
+  Services map[string]*PortList `protobuf:"bytes,22,rep,name=services,proto3" json:"services,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+  // Health status of the workload，such as: "Healthy"，"Unhealthy"
+  Status WorkloadStatus `protobuf:"varint,17,opt,name=status,proto3,enum=istio.workload.WorkloadStatus" json:"status,omitempty"`
+}
+```
+
+## How kmesh subscribe and use the workload model info
+
+1. kmesh's xds client subscribes to the workload model from Istiod through Delta manner, and the type_url is: "type.googleapis.com/istio.workload.Address";
+2. The workload data of Address type is divided into two sub resources: Service and Workload, which are responsed to kmesh and then parsed and converted into internal structures, stored in the bpf map;
+
+![kmesh subscribe workload](./kmesh_subscribe_workload.png)
+
+In the subsequent traffic management of kmesh, based on the IP and Port accessed by the client, find the corresponding service and its endpoints from the BPF map, and then randomly select an endpoint to forward the request to that endpoint;
```

Comment:
randonly select an endpoint is just current behavior, we should support more LB algorithms